diff -Nrup a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-02-25 15:20:19.000000000 +0100
+++ b/CMakeLists.txt	2025-02-27 17:58:15.702842131 +0100
@@ -716,34 +716,9 @@ if(AVIF_BUILD_APPS)
     endif()
     target_link_libraries(avifdec avif_apps avif avif_enable_warnings)
 
-    include(LocalLibargparse)
-
-    set(AVIFGAINMAPUTIL_SRCS
-        apps/avifgainmaputil/avifgainmaputil.cc
-        apps/avifgainmaputil/convert_command.cc
-        apps/avifgainmaputil/combine_command.cc
-        apps/avifgainmaputil/extractgainmap_command.cc
-        apps/avifgainmaputil/imageio.cc
-        apps/avifgainmaputil/printmetadata_command.cc
-        apps/avifgainmaputil/tonemap_command.cc
-        apps/avifgainmaputil/program_command.cc
-        apps/avifgainmaputil/swapbase_command.cc
-    )
-
-    add_executable(avifgainmaputil "${AVIFGAINMAPUTIL_SRCS}")
-    if(WIN32)
-        if(MSVC)
-            target_sources(avifgainmaputil PRIVATE apps/utf8.manifest)
-        elseif(MINGW)
-            target_sources(avifgainmaputil PRIVATE apps/utf8.rc)
-        endif()
-    endif()
-    set_target_properties(avifgainmaputil PROPERTIES LINKER_LANGUAGE "CXX")
-    target_link_libraries(avifgainmaputil libargparse avif_apps avif avif_enable_warnings)
-
     if(NOT SKIP_INSTALL_APPS AND NOT SKIP_INSTALL_ALL)
         install(
-            TARGETS avifenc avifdec avifgainmaputil
+            TARGETS avifenc avifdec
             RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
             ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
             LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
